### PR TITLE
Integrate gutenberg-mobile release 1.43.0

### PR DESCRIPTION
**Pull request by jd-alexander**
## Description
This PR incorporates the 1.43.0 release of gutenberg-mobile.  
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2874

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jd-alexander** [comment]
This PR can be merged as the lint failure [is not related](https://github.com/wordpress-mobile/WordPress-Android/pull/13466#issuecomment-738725776) to this PR. A fix for this issue is [currently in review.](https://github.com/wordpress-mobile/WordPress-Android/pull/13592)

**Review by cameronvoell — APPROVED** [review]
approved via wordpress-mobile/gutenberg-mobile#2874 (review). Should be ready to merge once we update the gutenberg-mobile ref to release merge on trunk

**Comment by guarani** [comment]
@hypest mentioned offline that it would be best to merge this since the linter error here happens on `develop` already and we'd rather merge this before any other PRs land on `develop`. I agree this sounds like a good idea.
I don't have access to merge this so would you mind merging @hypest?

**Pull request merged**
